rules-tests/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector/FixturePhp80/add_default_false_to_bool.php.inc

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Rector\Tests\Php80\Rector\Class_\ClassPropertyAssignToConstructorPromotionRector\FixturePhp80;
+
+final class AddDefaultFalseToBool
+{
+    public $name;
+
+    public function __construct(string $name = false)
+    {
+        $this->name = $name;
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\Php80\Rector\Class_\ClassPropertyAssignToConstructorPromotionRector\FixturePhp80;
+
+final class AddDefaultFalseToBool
+{
+    public function __construct(public string|false $name = false)
```

Comment:
In what rule? This rule only handles merge with default value. It's already tested on PHP 8.0, so it can't use `null` or `false` as standalone types.

Reply:
it seems current `ReturnUnionTypeRector`,

on config php 8.1, works as `?bool` as expected https://getrector.com/demo/dae6be7c-867a-4a25-a68d-967507a36109

on config php 8.2, it no change: https://getrector.com/demo/4bc78439-e92e-4433-a10f-0fbd58f5f523

Reply:
Fixtures added and covered :+1: